### PR TITLE
fix(callforwarding): changes to implement callforwarding for UCM users

### PR DESCRIPTION
**Pull request by sokn-sys**
# COMPLETES # https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-497087
https://jira-eng-gpk2.cisco.com/jira/browse/CALL-130499

## This pull request addresses
To get the callforward status match the directory number with either of the numbers (dn or e16number)

## Problem statement: 
The DN number returned in the call forward API response differs from the directory number we are passing as a parameter. We are comparing these two numbers, and due to the mismatch, we are experiencing issues retrieving the call forward status.

## by making the following changes

Display the callforward status for UCMC user on the MS Teams WxCaliing integration application
test

sdk_voicemail
sdk_destination

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]
The changes look good. Please add a test where cfa is assigned as the e164Number when dn is not present.

**Review by karthikraji2020 — CHANGES_REQUESTED** [review]
@sokn-sys could you please provide more context to this PR?

**Comment by sokn-sys** [comment]
> The changes look good. Please add a test where cfa is assigned as the e164Number when dn is not present.

Addressed the review comments

**Comment by sokn-sys** [comment]
> @sokn-sys could you please provide more context to this PR?

Updated the context and added one more Jira for reference.

File: packages/calling/src/CallSettings/UcmBackendConnector.test.ts [hunk 1]
```diff
@@ -95,6 +104,21 @@ describe('Call Settings Client Tests for UcmBackendConnector', () => {
       });
     });
 
+    it('Success: Get Call Forward Always setting when directory num matchcing with e16number and set to destination', async () => {
```
Comment (sreenara) [inline packages/calling/src/CallSettings/UcmBackendConnector.test.ts:107]: Can we fix the typo in the word matching?

**Review by sreenara — APPROVED** [review]
Small comment

**Review by Shreyas281299 — APPROVED** [review]
LGTM

**Review by karthikraji2020 — APPROVED** [review]
changes LGTM

**Pull request merged**